osu.Game.Rulesets.Osu/Difficulty/Evaluators/SpeedEvaluator.cs

```diff
@@ -54,12 +55,23 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current)
             double speedBonus = 1.0;
 
             if (strainTime < min_speed_bonus)
-                speedBonus = 1 + 0.75 * Math.Pow((min_speed_bonus - strainTime) / speed_balancing_factor, 2);
+                speedBonus = 1 + 0.70 * Math.Pow((min_speed_bonus - strainTime) / speed_balancing_factor, 2);
 
+            double angleBonus = calculateAngleBonus(osuCurrObj.Angle ?? Math.PI);
             double travelDistance = osuPrevObj?.TravelDistance ?? 0;
-            double distance = Math.Min(single_spacing_threshold, travelDistance + osuCurrObj.MinimumJumpDistance);
+            double distance = (travelDistance + osuCurrObj.MinimumJumpDistance) * angleBonus;
+            double distanceBonus = distance_multiplier * Math.Min(1, Math.Pow(distance / single_spacing_threshold, 3.5));
 
-            return (speedBonus + speedBonus * Math.Pow(distance / single_spacing_threshold, 3.5)) * doubletapness / strainTime;
+            double speedDifficulty = (speedBonus + distanceBonus) * doubletapness / strainTime;
+
+            return speedDifficulty;
+        }
+        private static double calculateAngleBonus(double angle)
```

Comment:
The idea of increasing distance by angle means that acute angles in streams are considered to be flowed in a circular-ish way (which they usually aren't except rare potentially-snappable cases), therefore this angle bonus makes distance evaluation abusable (you can make a pattern that requires way smaller path than evaluated) and I think that's not the direction we should go.

Reply:
I'm now using your angular velocity solution to this